src/axom/sidre/spio/IOManager.cpp

```diff
@@ -173,26 +181,26 @@ void IOManager::write(sidre::Group* datagroup, int num_files,
     m_baton = new IOBaton(m_mpi_comm, num_files, m_comm_size);
   }
 
-  std::string root_string = file_string;
-#ifdef AXOM_USE_SCR
-  if (m_use_scr)
-  {
-    SCR_Start_checkpoint();
+  if (m_use_scr && num_files != m_comm_size) {
+    SLIC_WARNING("SCR requires a file per process");
```

Comment:
This is no longer technically required either.  As part of this work, I modified SCR to allow for shared files.  If the user wants multiple processes to share access to a file, they must configure SCR to read/write those datasets on the parallel file system where all procs can see them.  However, if they are using file-per-process, then they can configure SCR to store files in node-local storage instead.  So the user can gain performance when using a file-per-process, but they can still use SCR for shared files.

Reply:
I'm going to leave this for now, as all of the SCR implementation here had the expectation to only work in file-per-process mode, so there isn't an immediate need to change this.  If we get everything to a state that we're happy with, we can think about loosening this restriction.

Reply:
@kennyweiss , from the SCR side, we can now support shared file access for both checkpoint (write) and restart (read).  If the user wants to use shared files (e.g., ``num_files < m_comm_size``), SCR must be configured to direct files to the parallel file system so that all procs can see all files.  If the user is running with file-per-process, they then have the option to configure SCR use node-local storage like ramdisk or an ssd, which gives a speed boost and enables things like background transfers.

Our plan right now is to have SCR_Route_file point at the parallel file system by default.  To use cache, the user will need to take some extra steps to configure SCR to point at cache *and* they will have to be sure they are using file-per-process in SPIO.

I have some text about that here:  https://github.com/LLNL/axom/pull/251#issuecomment-631817064

So it sounds like we should perhaps change this to an error, or perhaps it could be removed?  Which way do you guys want to go with it?

Reply:
Thanks @adammoody.

My first take is that we should keep this as an error for now, and create a ticket to rethink/rework this in the future.
Thoughts: @nselliott  or others?

Reply:
I agree with @kennyweiss, this should be an error for now.  Let's get the file-per-process mode in a good state and then add more flexibility in the future.  Since we don't have an alternate code path for when this check fails, it's better to be an error than a warning.

Reply:
I changed this from a warning to an error